File: tests/lib/env.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ENV_PATH = path.resolve(__dirname, "..", "..", ".env");

// Minimal .env parser (KEY=VALUE per line, '#' comments, blank lines
// ignored) -- no dependency added just to read one optional local secrets
// file. Never throws: a missing .env is the normal case (this repo's own
// CI/most contributors' machines don't have one), so callers always get an
// object back, just possibly without the key they asked for.
function parseEnvFile(filePath) {
  const out = {};
  let text;
  try {
    text = fs.readFileSync(filePath, "utf8");
  } catch (err) {
    return out;
  }
  for (const rawLine of text.split("\n")) {
    const line = rawLine.trim();
    if (!line || line.startsWith("#")) continue;
    const eq = line.indexOf("=");
    if (eq === -1) continue;
    const key = line.slice(0, eq).trim();
    let value = line.slice(eq + 1).trim();
    if ((value.startsWith('"') && value.endsWith('"')) || (value.startsWith("'") && value.endsWith("'"))) {
      value = value.slice(1, -1);
    }
    if (key) out[key] = value;
  }
  return out;
}

// process.env wins over .env, so a real CI secret (if this ever runs
// there) is never shadowed by a stray local file.
export function loadEnvKey(name) {
  if (process.env[name]) return process.env[name];
  const parsed = parseEnvFile(ENV_PATH);
  return parsed[name] || null;
}

File: ontology_translation/tools/env.py
"""Local secrets loader for ontology_translation/tools/*.py.

Mirrors tests/lib/env.mjs's convention exactly, so the two credential-loading
paths in this repo (JS live tests, Python translation tools) behave
identically: read the real environment first, fall back to a hand-parsed
`.env` at the repo root, and never throw just because that file is absent
(the normal case on most contributors' machines and always the case in CI).
No dependency added (no python-dotenv) just to read one optional file.
"""

from __future__ import annotations

import os
from pathlib import Path

REPO_ROOT = Path(__file__).resolve().parents[2]
ENV_PATH = REPO_ROOT / ".env"


def _parse_env_file(path: Path) -> dict:
    out = {}
    try:
        text = path.read_text(encoding="utf-8")
    except OSError:
        return out
    for raw_line in text.splitlines():
        line = raw_line.strip()
        if not line or line.startswith("#"):
            continue
        if "=" not in line:
            continue
        key, _, value = line.partition("=")
        key = key.strip()
        value = value.strip()
        if len(value) >= 2 and value[0] == value[-1] and value[0] in "\"'":
            value = value[1:-1]
        if key:
            out[key] = value
    return out


def load_env_key(name: str, default: str | None = None) -> str | None:
    """process.env-equivalent wins over .env, matching tests/lib/env.mjs."""
    if os.environ.get(name):
        return os.environ[name]
    parsed = _parse_env_file(ENV_PATH)
    return parsed.get(name, default)


def load_azure_config() -> dict:
    """Collects the four Azure OpenAI settings these tools need.

    Returns a dict with possibly-None values rather than raising, so callers
    (compile.py's --dry-run path in particular) can run without credentials
    and only fail at the point a real API call is actually attempted.
    """
    return {
        "endpoint": load_env_key("AZURE_OPENAI_ENDPOINT"),
        "api_key": load_env_key("AZURE_OPENAI_API_KEY"),
        "api_version": load_env_key("AZURE_OPENAI_API_VERSION", "2024-12-01-preview"),
        "deployment": load_env_key("AZURE_OPENAI_DEPLOYMENT", "gpt-5.4"),
    }
